api/net/Areas/Admin/Controllers/ReportController.cs

```diff
@@ -236,11 +235,13 @@ public async Task<IActionResult> Publish(int id)
     [SwaggerOperation(Tags = new[] { "Report" })]
     public async Task<IActionResult> Preview(ReportModel model)
     {
-        var subjectTemplateText = model.Settings.GetDictionaryJsonValue<string>("subject") ?? "";
-        var subjectTemplate = _templateEngine.AddOrUpdateTemplateInMemory($"report-{model.Id}-subject", subjectTemplateText);
-        var bodyTemplate = _templateEngine.AddOrUpdateTemplateInMemory($"report-{model.Id}", model.Template);
+        if (model.Template == null) throw new InvalidOperationException("Report template is missing from model");
+        var subjectTemplate = _templateEngine.AddOrUpdateTemplateInMemory($"report-{model.Id}-subject", model.Template?.Subject ?? throw new InvalidOperationException("Report subject template must be provided."));
```

Comment:
We now have a dedicated database field for the subject template.  It used to be within the `Settings` JSON property.